### DISCUSSION TITLE
Unable to get facebook to authenticate...

### DISCUSSION THREAD

**Discussion by micrb** [nextauthjs/next-auth#3733] [Help]
### Question 💬

I am having some trouble getting facebook to authenticate.  I am using standard nextjs-auth.  I get the following error:

<img width="590" alt="Screenshot 2022-01-22 at 16 26 01" src="https://user-images.githubusercontent.com/5666707/150647136-08f6346e-fd41-4649-b7c0-2589e2e62ec9.png">

I am using localhost:3000 and my setup for facebook login is as follows:
<img width="500" alt="Screenshot 2022-01-22 at 16 31 38" src="https://user-images.githubusercontent.com/5666707/150647211-6852926b-00e2-4460-be9e-4d45cf86392b.png">

my understanding is facebook login does not require redirect url's for localhost? I tried anyway but the field doesn't accept localhost unless you add httpS://localhost

Node v16.13.2
npx v8.1.2
yarn 1.22.17

I created this project with yarn create and the app is standard.

Please let me know if you need any more details.  I have tried various difference combinations of trying to add localhost, facebook, the redirect url into various domains in the facebook app, but nothing I have tried seems to work.

I am using firebase as a backend and I went as far as enabling facebook authentication on their to give me a redirect url (as localhost doesn't work), but that also didn't work either.

I imagine it's something very silly I am missing.

### How to reproduce ☕️

Create a nextjs app with yarn, add nextjs-auth, create a facebook app.

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this.

**Comment by viperxeyes** [comment]
Make sure you are trying to login using your developer account , if you use any other account to login in localhost you will get this error .

BUT for me it started 2 days ago even with developer account  logging on localhost i get expected 200 OK, got: 400 Bad Request, name: 'OAuthCallbackError'

**Comment by micrb** [comment]
> Make sure you are trying to login using your developer account , if you use any other account to login in localhost you will get this error .
> 
> BUT for me it started 2 days ago even with developer account logging on localhost i get expected 200 OK, got: 400 Bad Request, name: 'OAuthCallbackError'

Fore sure, I'm using the test account generated in the settings.

**Reply by viperxeyes** [reply]
Don't use the test account that has been generated , use the facebook account you registered your app with facebook.

**Comment by dominik-sfl** [comment]
Any news on this? I  am currently dealing with same issue? Any idea what this could be related to? 

edit: I just figured out that I had created a "Business" type app instead of a "Consumer" one. When using next-auth with a Consumer type app, it works for me. However, I do want to use a Business type app, as I want to use Instagrams GraphAPI. 

![image](https://user-images.githubusercontent.com/86602310/152543873-3a20d96d-3cec-4c8c-9e58-5c38cd3518a0.png)

Should I be able to use next-auth for this? Or does it only work with Consumer type apps?

**Reply by viperxeyes** [reply]
It does work with any kind of app you choose, when you register for facebook developer any login attempt with any account other than the developer account will result in this error.

example: you sign up for facebook with 1234@facebook.com , then you go to facebook developer and register for 1234@facebook.com ,you add the facebook login to your app , now you won't be able to to use facebook login except with 1234@facebook.com , any other account will result in this error page , unless you move to production by ticking the switch button in the app setting which require verfication and other stuff.

**Reply by dominik-sfl** [reply]
Thanks for your reply, but I don't think that's it in my case. 

Been using my developer account to login, I haven't even created other accounts.

**Reply by mingderwang** [reply]
me too, not in my case

**Comment by Jdelbarcogarza** [comment]
I'm dealing with the same issue. Is it possible to use the NextAuth to login to Facebook and from there use the marketing API from Facebook? To use the marketing API I forcefully need to create an app of the type Business by the way.

**Reply by tarcisiopqdt** [reply]
I want to do the same, did you do it? How?

**Comment by OzanKurt** [comment]
Still no luck?

**Comment by emperorwurm** [comment]
https://yourdomain.com/api/auth/callback/facebook  <--- change yourdomain.com to your production domain.. and put that into the allowed URI on facebook settings....

**Reply by nattadons** [reply]
"Can users log in with Facebook if my app hasn't passed App Review, even if it's in Live Mode?"

I have an issue with Facebook login. I can log in with my own Facebook account because I am the owner of the Facebook Developer Console and have access to the app's credentials (App ID and Secret). However, when trying to log in with other Facebook accounts, the login fails. My app is in Live Mode, but it hasn't passed App Review yet.

Is it possible to log in with Facebook using other accounts in this case? If not, what steps do I need to take to enable login for all users? Any insights or advice on how to fix this would be much appreciated!

Thanks in advance!


-
